pkg/kritis/pods/pods.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pods
+
+import (
+	"encoding/json"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	patch "k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Pods returns a list of pods in a namespace
+func Pods(namespace string) ([]corev1.Pod, error) {
+	clientset, err := getClientSet()
+	if err != nil {
+		return nil, err
+	}
+	pods, err := clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+	return pods.Items, err
+}
+
+// Images returns a list of images in a pod
+func Images(pod corev1.Pod) []string {
+	images := []string{}
+	for _, ic := range pod.Spec.InitContainers {
+		images = append(images, ic.Image)
+	}
+	for _, c := range pod.Spec.Containers {
+		images = append(images, c.Image)
+	}
+	return images
+}
+
+// applyPatch applies the patches in the modified pod to the running pod
+func applyPatch(modifiedPod *corev1.Pod, originalJSON []byte) error {
```

Comment:
Please add tests for this.
The json patch generated is easy to verify since its something like this.
https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#alternate-forms-of-the-kubectl-patch-command

Reply:
For testing refactor out generate patch which will return the patch.